components/user-service/src/main/java/edu/jhuapl/dorset/fileuserservice/FileUserService.java

```diff
@@ -0,0 +1,139 @@
+/*
+ * Copyright 2016 The Johns Hopkins University Applied Physics Laboratory LLC
+ * All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package edu.jhuapl.dorset.fileuserservice;
+
+import java.io.File;
+import java.io.FileInputStream;
+import java.io.FileOutputStream;
+import java.io.IOException;
+import java.io.OutputStream;
+import java.util.Enumeration;
+import java.util.HashMap;
+import java.util.Map;
+import java.util.Properties;
+import java.util.Set;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import edu.jhuapl.dorset.users.User;
+import edu.jhuapl.dorset.users.UserException;
+import edu.jhuapl.dorset.users.UserService;
+
+public class FileUserService implements UserService {
+    private static final Logger logger = LoggerFactory.getLogger(FileUserService.class);
+
+    private Map<String, User> users;
+    protected String userDirPath;
+    protected String fileBaseName;
+
+    /**
+     * 
+     * FileUserService
+     * 
+     * @param userDirectoryPath Path to directory that contains User files
+     * @param fileBaseName Base string of User file names
+     */
+    public FileUserService(String userDirectoryPath, String fileBaseName) {
+        users = new HashMap<String, User>();
+        this.userDirPath = userDirectoryPath;
+        this.fileBaseName = fileBaseName;
+    }
+
+    @Override
+    public String create(User user) throws UserException {
+        Properties prop = new Properties();
+        OutputStream output = null;
+        String userFilePath = this.userDirPath + this.fileBaseName + "-" + user.getUserName()
+                        + ".properties";
+
+        // Check if Username already exists
+        if (new File(userFilePath).exists()) {
+            throw new UserException(user.getUserName());
+        }
+
+        try {
+            output = new FileOutputStream(userFilePath);
+
+            Set<String> userKeys = user.getUserInformationKeys();
+            for (String key : userKeys) {
+                prop.setProperty(key, user.getUserInformation(key));
+            }
+            prop.store(output, null);
+
+        } catch (IOException e) {
+            throw new UserException("Username (" + user.getUserName() + ") already exists.");
```

Comment:
This message does not fit the code. I think a failure here means there was an error opening the file or writing to it (not having permission to write or a full disk are 2 possible causes).